recipes/cc65/all/conanfile.py

```diff
@@ -0,0 +1,164 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, MSBuild, tools
+from conans.client.tools.win import msvs_toolset
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class Cc65Conan(ConanFile):
+    name = "cc65"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://cc65.github.io/"
+    description = "A freeware C compiler for 6502 based systems"
+    license = "Zlib"
+    topics = ("conan", "cc65", "compiler", "cmos", "6502", "8bit")
+    exports_sources = "patches/**"
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    _autotools = None
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.arch not in ("x86", "x86_64"):
+                raise ConanInvalidConfiguration("Invalid arch")
+            if self.settings.arch == "x86_64":
+                self.output.info("This recipe will build x86 instead of x86_64 (the binaries are compatible)")
```

Comment:
I think we'll need to come back to this at some point in time. We cannot do this right now and it is something we are talking about here (https://github.com/conan-io/conan/pull/6952). This is the perfect example of a recipe that won't compile in `x86_64` but is compatible with other settings.

We would like to raise a `ConanInvalidConfiguration` here for `x86_64` (we don't want to build it), but we would love to fallback to the `x86` binary package (`compatible_packages` ©️ ) if we are going to run the binary...

It looks like that the workaround would be to raise the exception in the `build()` method, but we cannot do that because the backend of ConanCenter checks for `ConanInvalidConfiguration` before trying to build the libraries.

So, don't change this, but I really hope that we will able to iterate and improve this recipe. Thanks for giving us a real example! \o/